crates/datastore/src/traits.rs

```diff
@@ -186,6 +188,24 @@ pub struct TxData {
     // TODO: Store an `Arc<String>` or equivalent instead.
 }
 
+/// A record of a list of deletes for and potential truncation of a table,
+/// within a transaction.
+pub struct TxDeleteEntry {
+    /// Were all rows previously in the table deleted within this transaction?
+    truncated: TxTableTruncated,
+    /// The deleted rows of the table.
+    rows: Arc<[ProductValue]>,
+}
+
+/// Whether a table was truncated in a transaction.
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Debug)]
+pub enum TxTableTruncated {
+    /// It was truncated.
+    Yes,
+    /// It was not truncated.
+    No,
+}
```

Comment:
Storing the rows for a truncated table is surprising to me. Why is that necessary?